cmd/otelcol/main.go

```diff
@@ -176,32 +167,62 @@ func checkRuntimeParams() {
 	setMemoryLimit(memTotalSizeMiB)
 }
 
-// Validate and set the configuration
-func setConfig() {
-	// Check if the config is specified via the env var.
-	config := os.Getenv(configEnvVarName)
-	// If not attempt to use a default config; supports Docker and local
-	if config == "" {
+// Validate and equate specified config file path flag to the config file path env var
+func setConfigSource() {
+	// Config file path from cmd flag --config.
+	pathFlag := getKeyValue(os.Args[1:], "--config")
+	// Config file path from env var.
+	pathVar := os.Getenv(configEnvVarName)
+	// Config YAML from env var.
+	yamlVar := os.Getenv(configYamlEnvVarName)
+
+	// Restricting specifying config file path and config YAML env vars simultaneously.
+	if pathVar != "" && yamlVar != "" {
+		log.Fatalf("Specifying env vars %s and %s simultaneously is not allowed", configEnvVarName, configYamlEnvVarName)
```

Comment:
If pathFlag is provided I'm not sure this matters, but believe we'd want to clear both env vars.